parser/parser.y

```diff
@@ -2943,6 +2943,14 @@ FunctionCallKeyword:
 	{
 		$$ = &ast.FuncCallExpr{FnName: model.NewCIStr($1), Args: $3.([]ast.ExprNode)}
 	}
+|	"DATE" stringLit
```

Comment:
We may need a function to check whether stringLit is a valid input.

In MySQL:
``` sql
mysql> select date'2017-99-99';
ERROR 1525 (HY000): Incorrect DATE value: '2017-99-99'

mysql> select date'2017-2-31';
ERROR 1525 (HY000): Incorrect DATE value: '2017-2-31'

mysql> select date'201712-12';
ERROR 1525 (HY000): Incorrect DATE value: '201712-12'

mysql> select date'20171212';
+----------------+
| date'20171212' |
+----------------+
| 2017-12-12     |
+----------------+
1 row in set (0.01 sec)

mysql> select date'2017/12/12';
+------------------+
| date'2017/12/12' |
+------------------+
| 2017-12-12       |
+------------------+
1 row in set (0.00 sec)

mysql> select date'2017/12-12';
+------------------+
| date'2017/12-12' |
+------------------+
| 2017-12-12       |
+------------------+
1 row in set (0.00 sec)
```